File: Python/477. Total Hamming Distance.py
class Solution:
    def totalHammingDistance(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans, n = 0, len(nums)
        for i in range(32):
            tmp = 0
            for num in nums:
                tmp += ((num>>i)&1)
            ans += tmp * (n-tmp)
        return ans

File: C++/477. Total Hamming Distance.cpp
class Solution {
public:
    int totalHammingDistance(vector<int>& nums) {
        int ans = 0;
        for(int i=0;i<32;i++)
        {
            int tmp = 0;
            for(auto &num: nums)
            {
                if((num>>i)%2==1)
                    tmp++;
            }
            ans += tmp*(nums.size()-tmp);
        }
        return ans;
    }
};